### PR TITLE
Prefill data from onboarding for in case of non-progressive Stripe KYC

### LINKED ISSUE #5476: Submit and Redirect to Stripe
### Description

We will need to adjust our existing logic, currently we pass a progressive flag from the client, but this will need to change since the intelligent router will decide whether to pass the user to PO or not. This issue handles any changes required to the Stripe redirect performed by the onboarding form.

### Acceptance Criteria

* We appropriately redirect users to either the normal Stripe KYC, or the PO KYC, with values prefilled.

### PR DESCRIPTION
**Pull request by oaratovskyi**
Fixes #5476 

### Changes proposed in this Pull Request

This PR tweaks the client's PHP side to support data prefill for non-progressive Stripe KYC scenario.

#### Pre-requisites
- Check that Progressive Onboarding WCPay Dev Tools checkbox is checked
- Check out branch `add/5476-prefill-fields-for-full-stripe-kyc`

#### Progressive onboarding, individual
- Fill in the form before Stripe KYC
    - first name, last name, email, phone (+10000000000)
    - legal name of the business, business website (http://wcpay.test), country (US), legal entity (individual)
    - Annual revenue: Less than 250k, time to go live: within 1 month
- You should be redirected to Let’s get you set up for payments screen, then to PO KYC with the next data prefilled:
    - Country
    - Business Type
    image
    - First Name
    - Last Name
    - Email
    image
    - Business URL (you won't be asked to type it)
    - Business name (you won't be asked to type it)
    - MCC is currently hard-coded, because the client does not pass it, should be fixed after testing
- Click Agree and Submit and check that all data (that you filled in and that was not prompted) is in Stripe (doing business as, business URL, MCC). 

#### Full onboarding, individual
- Fill in the form before Stripe KYC
    - first name, last name, email, phone (+10000000000)
    - legal name of the business, business website (http://wcpay.test), country (US), legal entity (individual)
    - Annual revenue: 250k - 1M, time to go live: 1-3 months
- You should be redirected to Let’s get you set up for payments screen, then to Full KYC with the next data prefilled:
    - Country
    - Business Type
   image
    - First Name
    - Last Name
    - Email
    image
    - Business URL (you won't be asked to type it)
    - Business name (you won't be asked to type it)
    - Phone (you won't be asked to type it)
    - MCC is currently hard-coded, because the client does not pass it, should be fixed after testing
- Fill in Bank account info, Click Agree and Submit and check that all data (that data you filled in and that was not prompted) is in Stripe (doing business as, business URL, MCC). The phone is sent to Stripe but I didn't find where it can be found in the Stripe dashboard.

#### Progressive onboarding, company
- Fill in the form before Stripe KYC
    - first name, last name, email, phone (+10000000000)
    - legal name of the business, business website (http://wcpay.test), country (US), legal entity (company, sole prop)
    - Annual revenue: Less than 250k, time to go live: within 1 month
- You should be redirected to Let’s get you set up for payments screen, then to PO KYC with the next data prefilled:
    - Country
    - Business Type (with subcategory)
    image
    - Doing business as 
    image
    - First Name, Last Name can't be prefilled so far, there is a discussion in p1680091022120479-slack-C9976E5MJ
    - Email
    image
    - Business URL (you won't be asked to type it)
    - MCC is currently hard-coded, because the client does not pass it, should be fixed after testing
- Click Agree and Submit and check that all data (that you filled in and that was not prompted) is in Stripe (business URL, MCC). 
- 
#### Full onboarding, company
- Fill in the form before Stripe KYC
    - first name, last name, email, phone (+10000000000)
    - legal name of the business, business website (http://wcpay.test), country (US), legal entity (company, sole prop)
    - Annual revenue: 250k - 1M, time to go live: 1-3 months
- You should be redirected to Let’s get you set up for payments screen, then to PO KYC with the next data prefilled:
    - Country
    - Business Type (with subcategory)
    image
    - Doing business as
    - I wasn't successful in prefilling business support phone. It requires server changes and even when I did it, it didn't work. I guess I'll create a separate issue for that.
    image
    - First Name, Last Name can't be prefilled so far, there is a discussion in p1680091022120479-slack-C9976E5MJ
    - Email
    image
    - Business URL (you won't be asked to type it)
    - MCC is currently hard-coded, because the client does not pass it, should be fixed after testing
- Fill remaining steps and click Agree and Submit and check that all data (that you filled in and that was not prompted) is in Stripe (business URL, MCC). 

#### Current onboarding
- Un-check  Progressive Onboarding WCPay Dev Tools checkbox
- Onboard using our current KYC and check that nothing is prefilled (both for individual and company)
- Check that onboarding still works properly

-------------------

- [x] Run `npm run changelog` to add a changelog file, and choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5857` or branch name `add/5476-prefill-fields-for-full-stripe-kyc` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5476-prefill-fields-for-full-stripe-kyc) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7e47ae82bc247efcbecabdcf5e2e2bbe7ab81f87
- Build time: 2023-03-31 09:17:47 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/onboarding-prototype/steps/loading.tsx [hunk 1]
```diff
@@ -58,7 +56,9 @@ const Loading: React.FC = () => {
 			? addQueryArgs( connectUrl, {
 					progressive: fromDotNotation( data ),
 			  } )
-			: connectUrl;
+			: addQueryArgs( connectUrl, {
+					prefill: fromDotNotation( data ),
+			  } );
 		window.location.href = resultUrl;
```
Comment (ismaeldcom) [inline client/onboarding-prototype/steps/loading.tsx:59]: __Nitpick__: We can avoid writing`addQueryArgs` twice using a dynamic key. Something like:
```js
const resultUrl = addQueryArgs(connectUrl, {
  [isElegible ? 'prefill' : 'progressive']: fromDotNotation(data),
});

```

File: includes/class-wc-payments-account.php
Comment (ismaeldcom) [inline includes/class-wc-payments-account.php:1]: Now that the experiment is over, and with this new onboarding flow we will require every field to be filled, we can get rid of `$prefill_rest` definition and get it directly like what we do with `$progressive_onboarding`.

**Review by ismaeldcom — COMMENTED** [review]
Looks good! Thanks for taking care of it so quickly. I see a couple of things we can do to simplify the code a bit.

And thinking further on the new onboarding flow, I think we can simplify it a bit more. We will be sending the same data but in different props `business_data vs account_data`. Maybe sending it using the same prop but adding a boolean for progressive is better.

Anyway, that could be a nice to have issue to solve 🙂 

>the phone is somehow not pre-filled. I tried +10000000000 and 0000000000, but nothing works. Do you know what might be the issue?

AFAIK we are not sending that phone to Stripe, it's for us to have a contact phone number from the merchant. Stripe has `business_profile.support_phone` and `individual.phone`, but I'm not sure if we should send it there. 🤔 In test mode Stripe KYC don't ask for the phone number for 2FA, and doesn't look like the API has a phone field for that.

**Comment by oaratovskyi** [comment]
@ismaeldcom huge thanks for the review! Good point about the experiment being over 🙂 I thought we still need to support it, and that's why I was hesitant to refactor it. Now I'll follow your suggestions, all make sense and look good to me!

File: includes/class-wc-payments-account.php
Comment (oaratovskyi) [reply includes/class-wc-payments-account.php:1]: Fixed in https://github.com/Automattic/woocommerce-payments/pull/5857/commits/9d784e9e18ba42be752b58303dad303fba8286fe and https://github.com/Automattic/woocommerce-payments/pull/5857/commits/014dfc0199051ffd2ec6991f51f3e5c9da588f02

File: client/onboarding-prototype/steps/loading.tsx (hunk 1 above)
Comment (oaratovskyi) [reply client/onboarding-prototype/steps/loading.tsx:59]: Fixed in https://github.com/Automattic/woocommerce-payments/commit/9d784e9e18ba42be752b58303dad303fba8286fe!

**Comment by oaratovskyi** [comment]
>  We will be sending the same data but in different props business_data vs account_data. Maybe sending it using the same prop but adding a boolean for progressive is better.
Anyway, that could be a nice to have issue to solve 🙂

@ismaeldcom Solved it, please take a look and say WDYT 🙂 

> Stripe has business_profile.support_phone and individual.phone

I was able to prefill `individual.phone`, but had issues with `business_profile.support_phone`. It requires server changes in Account_Service::create_stripe_account and even when it's done, still didn't work. Since there is an ongoing discussion about prefilling first name and last name for company, we can do that as follow-up (I'll create an issue for that). What do you think?

**Review by ismaeldcom — APPROVED** [review]
Looks great! Thanks for those detailed testing instructions, everything works as expected. I have a small suggestion, but I think we can debate and have a follow-up on this if we decide to. So ready to be merged 🙂 

>Solved it, please take a look and say WDYT 🙂

Definitely looks better now! Although I was thinking on keeping the progressive way of sending everything to the server without mapping it on the client. With the idea of having our own structure and map it later on the server, based on our needs. Like what we've mentioned about the phone number or MCC.

Anyway, it's good to have this extra validation and data mapping. We can decide later if we prefer to stick to this structure or map it on the server 🙂 

>Since there is an ongoing discussion about prefilling first name and last name for company, we can do that as follow-up (I'll create an issue for that). What do you think?

Sounds good to me.

**Comment by oaratovskyi** [comment]
Thanks, @ismaeldcom! I've created a follow-up issue and described everything here #5929. Feel free to add any details!

**Pull request merged**